app/components/emoji/emoji.tsx

```diff
@@ -2,44 +2,45 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
 import {
     Platform,
+    StyleProp,
     StyleSheet,
     Text,
+    TextStyle,
 } from 'react-native';
-import FastImage from 'react-native-fast-image';
+import FastImage, {ImageStyle} from 'react-native-fast-image';
 
-export default class Emoji extends React.PureComponent {
-    static propTypes = {
+type Props = {
 
-        /*
+    /*
          * Emoji text name.
          */
-        emojiName: PropTypes.string.isRequired,
-
-        /*
-         * Image URL for the emoji.
-         */
-        imageUrl: PropTypes.string.isRequired,
-
-        /*
-         * Set if this is a custom emoji.
-         */
-        isCustomEmoji: PropTypes.bool.isRequired,
-
-        /*
-         * Set to render only the text and no image.
-         */
-        displayTextOnly: PropTypes.bool,
-        literal: PropTypes.string,
-        size: PropTypes.number,
-        textStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
-        unicode: PropTypes.string,
-        customEmojiStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
-        testID: PropTypes.string,
-    };
+    emojiName: string;
+
+    /*
+     * Image URL for the emoji.
+     */
+    imageUrl: string;
+
+    /*
+     * Set if this is a custom emoji.
+     */
+    isCustomEmoji: boolean;
+
+    /*
+     * Set to render only the text and no image.
+     */
+    displayTextOnly?: boolean;
+    literal?: string;
+    size?: number;
+    textStyle?: StyleProp<TextStyle>;
+    unicode?: string;
+    customEmojiStyle?: StyleProp<ImageStyle>;
+    testID?: string;
+}
 
+export default class Emoji extends React.PureComponent<Props> {
```

Comment:
is it too much to ask that we convert this to a functional component instead?

Reply:
No problem ;)